### PR TITLE
[@types/react-datepicker] Update types for v6.0

### PR DESCRIPTION
**Pull request by worg**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/Hacker0x01/react-datepicker/releases/tag/v6.0.0>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/react-datepicker/package.json
1. types/react-datepicker/react-datepicker-tests.tsx

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d13386695b80041fbd55dff26de94d104293d1d8
</p>

**Review by Anonymous4078 — APPROVED** [review]

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/react-datepicker/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e2e83535f609326fee77bed50b97ba01e0720a09
</p>

File: types/react-datepicker/index.d.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
-import * as Popper from "@popperjs/core";
+import { Middleware, Placement, UseFloatingOptions } from "@floating-ui/react";
 import { Locale } from "date-fns";
 import * as React from "react";
-import { Modifier, StrictModifierNames } from "react-popper";
+
+type PopperProps = Partial<Omit<UseFloatingOptions, "open" | "whileElementsMounted" | "placement" | "middleware">>;
 
 export interface CalendarContainerProps {
     className?: string | undefined;
```
Comment (seckie) [inline types/react-datepicker/index.d.ts:8]: According to https://github.com/Hacker0x01/react-datepicker/pull/4393/files#diff-bdf7d74fe2b1de77e4844501d2f292338d6b334b41806ecc917361f5a7afcca8, you should remove `show PopperArrow` and `arrowProps` props from `CalendarContainerProps`. This change was included in v5.0.0.

**Review by seckie — CHANGES_REQUESTED** [review]

File: types/react-datepicker/index.d.ts (hunk 1 above)
Comment (worg) [reply types/react-datepicker/index.d.ts:8]: done in last commit

**Review by seckie — APPROVED** [review]
LGTM

**Comment by worg** [comment]
Ready to merge

**Pull request merged**